src/PSR7/SpecFinder.php

```diff
@@ -74,37 +108,49 @@ public function findPathSpec(OperationAddress $addr) : PathItem
         return $pathSpec;
     }
 
-    /**
-     * @return Parameter[]
-     *
-     * @throws NoPath
-     */
-    public function findPathSpecs(OperationAddress $addr) : array
+    private function findPathSimple(OperationAddress $addr) : ?PathItem
     {
-        $spec = $this->findOperationSpec($addr);
+        return $this->openApi->paths->getPath($addr->path());
+    }
 
-        // 1. Collect operation-level params
-        $pathSpecs = [];
+    private function findPathUsingFinder(OperationAddress $addr) : ?PathItem
+    {
+        $finder  = new PathFinder($this->openApi, $addr->path(), $addr->method());
+        $results = $finder->search();
+        if (count($results) === 1) {
+            return $this->findPathSimple($results[0]);
+        }
 
-        foreach ($spec->parameters as $p) {
-            if ($p->in !== 'path') {
-                continue;
-            }
+        return null;
+    }
 
-            $pathSpecs[$p->name] = $p;
+    /**
+     * @throws NoCallback
+     */
+    private function findCallbackInOperation(CallbackAddress $addr, Operation $operation) : Operation
```

Comment:
minor but I suggest to keep current public-protected-private order. this make PR easier even if not reflected in automatic style guides

Reply:
done